backend/remote-state/azure/client.go

```diff
@@ -67,6 +70,17 @@ func (c *RemoteClient) Put(data []byte) error {
 	}
 
 	ctx := context.TODO()
+
+	if c.versioning {
+		snapshotInput := blobs.SnapshotInput{LeaseID: options.LeaseID}
```

Comment:
The original approach didn't use a specific lease ID (just `nil`). I'm not sure what the best approach is here.

Reply:
my understanding of the Azure API is that a Lease ID is required if one is configured (and will error if not) - so the tests should confirm this